src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrBinaryFileScanner.cs

```diff
@@ -72,12 +73,31 @@ private async Task<IEnumerable<string>> ScanAndParseResult(string[] args, string
 
         ret.ThrowIfFailed($"Failed to scan the {repoName} repository");
 
-        // We want to exlude empty files from the list. These empty files still have a few chracters that we want to ignore
+        // Git marks UTF 16 text files as binary, we want to exclude these
         return ret.StandardOutput
             .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries)
             .Where(line => line.StartsWith(BinaryFileMarker))
             .Select(line => line.Split('\t').Last())
-            .Where(file => new FileInfo(_vmrInfo.VmrPath / file).Length > 6);
+            .ToAsyncEnumerable()
+            .WhereAwait(async file => await IsNotUTF16(_vmrInfo.VmrPath / file))
```

Comment:
Is this `WhereAwait` executing in parallel? If not, I'd just create tasks and `WhenAll` them to make them run in parallel as here I believe we can get a nice benefit. Although, I'd throttle the amount of spawned processes at once when doing this.

Reply:
Looks like they're not. There are no docs for `Linq.Async` so I had to write my own testing. I don't think this something that needs to be run in parallel, since we're already doing it on a per repo basis, not the big list at the end.

Reply:
I checked with the Stopwatch, it's a few milliseconds at most

Reply:
Okay, thanks, then leave as is